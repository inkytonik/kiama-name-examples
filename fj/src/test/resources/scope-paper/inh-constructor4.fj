// ADAPTED FROM inheritance/constructors.spt:missing super argument
// FIX: Add fields to A
class A extends Object {
  Object f;
  Object g;
  A(Object f, Object g) {
    super();
    this.f = f;
    this.g = g;
  }
}
class B extends A {
  B(Object f) {
    super(f);
  }
}
