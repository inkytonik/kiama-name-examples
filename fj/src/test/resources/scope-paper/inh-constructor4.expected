src/test/resources/scope-paper/inh-constructor4.fj:4:10:info: f def: Object
  Object f;
         ^
src/test/resources/scope-paper/inh-constructor4.fj:5:10:info: g def: Object
  Object g;
         ^
src/test/resources/scope-paper/inh-constructor4.fj:6:12:info: f def: Object
  A(Object f, Object g) {
           ^
src/test/resources/scope-paper/inh-constructor4.fj:6:22:info: g def: Object
  A(Object f, Object g) {
                     ^
src/test/resources/scope-paper/inh-constructor4.fj:13:3:error: constructor args (Object f) should be the fields (Object f, Object g)
  B(Object f) {
  ^
src/test/resources/scope-paper/inh-constructor4.fj:13:12:info: f def: Object
  B(Object f) {
           ^
